### DISCUSSION TITLE
Whats happen with the `s` command? 😱

### DISCUSSION THREAD

**Discussion by dvogt23** [LazyVim/LazyVim#76] [General]
Why it was substitute with lazy? Thats bother me from the beginning 😆 wdyt guys? Look like its now for easy-jump to char ...

**Comment by folke** [comment]
`s` is substitute by default in Neovim. You can use `cl` or `xi` instead :)

`s` is mapped to Leap. See https://github.com/ggandor/leap.nvim

**Reply by dvogt23** [reply]
whaaat :(( how to disable this by default? `leap.add_default_mappings(true)` in lazy 🤔

**Reply by folke** [reply]
just disable leap or override its config and chose your own mappings

**Comment by thyttan** [comment]
Added an upvote on this as I wondered the same. Now I'll read up on Leap to see if I want to change my config or not.

**Comment by briandipalma** [comment]
Learn to use `leap`/`flit`, it's well worth it.

**Reply by dvogt23** [reply]
I prefer [hop.nvim](https://github.com/phaazon/hop.nvim) and let the default keymapping as it is, for not confusing myself, when working on different machine

**Comment by kyriacos** [comment]
i tried overriding the keymaps for leap and i couldn't get that to work. It worked for other plugins. The keys show up but as leap is lazy loaded it seems the keys don't actually apply.

I even copied the entire leam.nvim spec and just changed the keys and that didn't work. Tried returning a function instead of `{}` hoping it would override but that didn't work either. And i also tried `add_default_mappings(false)`.

Any suggestions? I ended up disabling it for now. Thanks!

**Reply by dvogt23** [reply]
I think its not possible to overwrite the default_mapping function in lazyvim, cause in full spec](https://www.lazyvim.org/plugins/editor#leapnvim) you have no possability to set this (`leap.add_default_mappings(true)`) to `false`. 

Think you have to disable it in lazy, and import it on your own.

**Reply by folke** [reply]
Add a spec like the below. Check the leap docs to create your custom keymaps:

```lua
  {
    "ggandor/leap.nvim",
    keys = false,
    config = function(_, opts)
      local leap = require("leap")
      -- add your own keymaps here
    end,
  },
```

**Reply by kyriacos** [reply]
@dvogt23 Thanks for the response. I did try disabling it and then adding the full spec. But that resulted in it just being disabled.

**Reply by kyriacos** [reply]
@folke thanks! i'll try the `keys = false` part as well today.

**Comment by scottming** [comment]
I noticed that when using lazy.vim, the highlight **after/before** the cursor disappears after pressing 's/S'. This is actually a perfect state. Can you tell me how this was achieved? I searched the entire repository and found no code related to the `cursor` or `leap.nvim`.

**Reply by scottming** [reply]
o, this is via tokionight.

**Reply by scottming** [reply]
curious how to archive that if I use another colorsheme

**Reply by dpetka2001** [reply]
It's the `LeapBackDrop` highlight group. You should set it in another theme to whatever color you want. Supposedly that theme provides support for the specific highlight group and functionality to customize it. For example, catppuccin supports it but rose-pine and kanagawa which are other 2 themes that I use don't support it. Or you could set it yourself, since it's `leap` that originally provides it.

**Reply by scottming** [reply]
thanks, I'll try that.

**Comment by netherstar** [comment]
default s (S) is useless
